packages/dataviews/src/types.ts

```diff
@@ -240,7 +245,7 @@ export interface NormalizedFilter {
 	/**
 	 * The list of options to pick from when using the field as a filter.
 	 */
-	elements: Option[];
+	elements?: Option[];
```

Comment:
The idea of NormalizedFilter (or NormalizedField) is that all props should exist, so code that deals with this type doesn't need to do checks for existance (we're normalized them before). In this case, why does it need to change to undefined? I haven't checked all the code, but I suppose we're using this to check something. Can't we check for size of the array?

Reply:
I changed it to optional because the `useFilters` doesn't normalize the `elements` prop, it's pass through as is so it could be undefined or an empty array. If that's the case, then it's a bug. I'll revert the type and fix the bug to ensure the elements are always defined. We can check the size.

https://github.com/Automattic/wp-calypso/blob/b1fa25b44c054a32b0b5f1b417aa2a885eddad25/packages/dataviews/src/components/dataviews-filters/index.tsx#L44

Reply:
Fixed `filter.elements` type and checked array size instead. 58f3f5e88c5087a51366db86806b14ec042c20a3